### PR TITLE
Log the cycle consistent front-end metrics

### PR DESCRIPTION
**Pull request by johnwlambert**
Log the cycle consistent front-end metrics to get better sense of what measurements actually going into MVO, after cycle consistency checks.

### CODE REVIEW

**Review by akshay-krishnan — APPROVED** [review]
Not directly related to the PR, but is there a good reason to call cycle consistency in the scene optimizer? Its an MVO-level detail I think, so maybe we should move it there?

File: gtsfm/scene_optimizer.py [hunk 1]
```diff
@@ -176,11 +176,26 @@ def create_computation_graph(
...
+
+        if gt_pose_graph is not None:
+            two_view_reports_dict_cycle_consistent = dask.delayed(_filter_dict_keys)(
+                dict=two_view_reports_dict, ref_dict=i2Ri1_graph_dict
+            )
+            metrics_graph_list.append(
+                dask.delayed(two_view_estimator.aggregate_frontend_metrics)(
+                    two_view_reports_dict_cycle_consistent,
+                    self._pose_angular_error_thresh,
+                    metric_group_name="frontend_summary_after_cycle_consistency",
```
[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:196]: that's better, i'll update

File: gtsfm/scene_optimizer.py [hunk 2]
```diff
@@ -176,11 +176,26 @@ def create_computation_graph(
...
+                )
+            )
 
         # TODO (johnwlambert): dump metrics after cycle consistency
```
[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:200]: yes : - )

[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:200]: just removed it

**Comment by johnwlambert** [comment]
> Not directly related to the PR, but is there a good reason to call cycle consistency in the scene optimizer? Its an MVO-level detail I think, so maybe we should move it there?

that's true. only thing is that we should pass the two view report dictionaries to MVO then as well, since we use it for the cycle consistent metric? maybe in a future PR?

File: gtsfm/scene_optimizer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py]: updated

**Comment by akshay-krishnan** [comment]
> > Not directly related to the PR, but is there a good reason to call cycle consistency in the scene optimizer? Its an MVO-level detail I think, so maybe we should move it there?
> 
> that's true. only thing is that we should pass the two view report dictionaries to MVO then as well, since we use it for the cycle consistent metric? maybe in a future PR?

Sure, add a TODO maybe? We lose track of these things.

**Comment by johnwlambert** [comment]
> > > Not directly related to the PR, but is there a good reason to call cycle consistency in the scene optimizer? Its an MVO-level detail I think, so maybe we should move it there?
> > 
> > 
> > that's true. only thing is that we should pass the two view report dictionaries to MVO then as well, since we use it for the cycle consistent metric? maybe in a future PR?
> 
> Sure, add a TODO maybe? We lose track of these things.

Sure, just added a TODO here: https://github.com/borglab/gtsfm/issues/289 @akshay-krishnan

**Comment by ayushbaid** [comment]
Can you add a snapshot of the metrics from skydio-32, and what potential issues they highlight? Would help document this change better before we merge it.

**Comment by johnwlambert** [comment]
Door-12: 

<p style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; font-size: 25px; font-family: Arial;">Frontend Summary</p>

Metric name | Value
-- | --
angular_err_threshold_deg | 5
num_total_image_pairs | 66
num_valid_image_pairs | 66
rotation_success_count | 66
translation_success_count | 64
pose_success_count | 64
num_all_inlier_correspondences_wrt_gt_model | 35
mean_rot3_angular_errors_deg | 0.359034
mean_trans_angular_errors_deg | 1.04154
mean_pose_errors_deg | 1.07022

<br class="Apple-interchange-newline">

**Comment by johnwlambert** [comment]
Skydio-32

<p style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; font-size: 25px; font-family: Arial;">Frontend Summary</p>

Metric name | Value
-- | --
angular_err_threshold_deg | 5
num_total_image_pairs | 496
num_valid_image_pairs | 446
rotation_success_count | 289
translation_success_count | 244
pose_success_count | 220
num_all_inlier_correspondences_wrt_gt_model | 39
mean_rot3_angular_errors_deg | 13.9265
mean_trans_angular_errors_deg | 33.1812
mean_pose_errors_deg | 37.2674

<br class="Apple-interchange-newline">

<p style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; font-size: 25px; font-family: Arial;">Cycle Consistent Frontend Summary</p>

Metric name | Value
-- | --
angular_err_threshold_deg | 5
num_total_image_pairs | 375
num_valid_image_pairs | 375
rotation_success_count | 289
translation_success_count | 234
pose_success_count | 220
num_all_inlier_correspondences_wrt_gt_model | 39
mean_rot3_angular_errors_deg | 7.93462
mean_trans_angular_errors_deg | 23.8729
mean_pose_errors_deg | 26.9242

<br class="Apple-interchange-newline">

**Comment by johnwlambert** [comment]
Skydio-8: Rot3 error  3.7 -> 2.3, Unit3 error 28 -> 17. Helps!

<p style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; font-size: 25px; font-family: Arial;">Frontend Summary</p>

Metric name | Value
-- | --
angular_err_threshold_deg | 5
num_total_image_pairs | 28
num_valid_image_pairs | 28
rotation_success_count | 23
translation_success_count | 22
pose_success_count | 21
num_all_inlier_correspondences_wrt_gt_model | 6
mean_rot3_angular_errors_deg | 3.68011
mean_trans_angular_errors_deg | 28.1091
mean_pose_errors_deg | 28.606

<br class="Apple-interchange-newline">

<p style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; font-size: 25px; font-family: Arial;">Cycle Consistent Frontend Summary</p>

Metric name | Value
-- | --
angular_err_threshold_deg | 5
num_total_image_pairs | 26
num_valid_image_pairs | 26
rotation_success_count | 23
translation_success_count | 22
pose_success_count | 21
num_all_inlier_correspondences_wrt_gt_model | 5
mean_rot3_angular_errors_deg | 2.27704
mean_trans_angular_errors_deg | 17.1475
mean_pose_errors_deg | 17.6826

<br class="Apple-interchange-newline">

**Comment by johnwlambert** [comment]
> Can you add a snapshot of the metrics from skydio-32, and what potential issues they highlight? Would help document this change better before we merge it.

@ayushbaid for sure -- just added a summary

**Review by ayushbaid — APPROVED** [review]

**Comment by johnwlambert** [comment]
Thanks @ayushbaid !

**Pull request merged**